src/react-components/ory/user-logout-card.spec.tsx

```diff
@@ -0,0 +1,41 @@
+import { expect, test } from "@playwright/experimental-ct-react"
+import { Locator } from "@playwright/test"
+import { UserLogoutCard } from "./user-logout-card"
+
+let component: Locator
+
+test.beforeEach(async ({ mount }) => {
+  component = await mount(
+    <UserLogoutCard
+      csrfToken="CSRF token"
+      action="https://example.com/logout"
+      challenge="logout challenge"
+    />,
+  )
+})
+
+test("submits when user clicks 'Yes'", async () => {
+  const [request] = await Promise.all([
+    component.page().waitForRequest("https://example.com/logout"),
+    component.locator("button", { hasText: /yes/i }).click(),
+  ])
```

Comment:
This can be racy (see [here](https://www.notion.so/Playwright-E2E-Testing-Ory-Network-a88cba00aa3c4e3982bfa302e78be006?pvs=4#1f7b786a20ca48fb8c432c59f267cc53) and [here](https://playwright.dev/docs/api/class-page#page-wait-for-request))
You should first create the promise, then trigger the action and then await the promise.
```suggestion
  const requestPromise = component.page().waitForRequest("https://example.com/logout")
  await component.locator("button", { hasText: /yes/i }).click()
  const request = await requestPromise
```

Reply:
My code is actually equivalent to the change you suggested.

Lets consider this: `await Promise.all([a, b])`, where `a` and `b` are promises. Then during the array construction, `a` will always be called before `b`.

So, `await Promise.all([a, b])` is equivalent to `const c = [a, b]; Promise.all(c)`, which is doing the right thing: Setting up the listener for the navigation first, then triggering the click event, then awaiting that both finish.

The only way I can see this race is if we were already on the destined URL. Otherwise I find this pattern more concise than what you suggested. WDYT?

Reply:
Hmm strange, in the bundled Playwright doc the recommendation is still to use `Promise.all`, but in the current version this was changed to the version you suggested. But I still can't see why this would race.

Reply:
I'm not sure, the order is always guaranteed like that. We actually had cases where this was causing flaky behaviour (though this was primarily with `waitForNaviagation` which is in itself problematic).
You are probably right that it is no problem in this case. The proposed pattern is guaranteed to work correctly though.

@jonas-jonas do you have some more insights into this?

Reply:
[here](https://github.com/microsoft/playwright/discussions/20779#discussioncomment-4942948) is a reason for the change in the documentation. They say there is no functional difference, but the newer pattern has fewer pitfalls for users.

Reply:
Ah, nice find. Then I'm happy, because my mental model of JS is still intact, and I'll still accept your suggestion because it has fewer pitfalls (which, in a language littered with pitfalls, is a strong argument).